j-lawyer-client/src/com/jdimension/jlawyer/client/editors/files/InvolvedPartyEntryPanel.java

```diff
@@ -1455,6 +1483,54 @@ private void mnuCopyActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST
         }
     }//GEN-LAST:event_mnuCopyActionPerformed
 
+    private void mnuFindRouteToAddressActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_mnuFindRouteToAddressActionPerformed
+        if (this.a != null) {
+            
+            // PROFILE as startAdress
+            ServerSettings sset = ServerSettings.getInstance();
+            String companyCity = sset.getSetting(ServerSettings.PROFILE_COMPANYCITY, "");
+            String companyStreet = sset.getSetting(ServerSettings.PROFILE_COMPANYSTREET, "");
+            String companyZip = sset.getSetting(ServerSettings.PROFILE_COMPANYZIP, "");
+            
+            String startAddress = companyCity + "+" + companyZip + "+" + companyStreet.replace(" ", "+");
+            String destinationAddress = a.getCity() + "+" + a.getZipCode() + "+" + a.getStreet() + "+" + a.getStreetNumber();
+            
+            // JOptionPane.showMessageDialog(null, startAddress);  
+            
+            try {
+                
+                String coordinates1 = FindRouteToAddress.getCoordinates(startAddress);
+                String coordinates2 = FindRouteToAddress.getCoordinates(destinationAddress);
+   
+                if (coordinates1 != null && coordinates2 != null) {
+                    
+                    FindRouteToAddress.openBrowserWithRoute(coordinates1, coordinates2);
+                    
+                } else {
+                    JOptionPane.showMessageDialog(null, "Die Koordinaten für eine oder beide Adressen konnten nicht abgerufen werden.");
```

Comment:
in an ideal world, error messages should be actionable and provide a hint for the user to fix the root cause. i reworked the code like this:

```
String coordinates1 = FindRouteToAddress.getCoordinates(startAddress);
                if(coordinates1==null) {
                    JOptionPane.showMessageDialog(this, "Koordinaten der Startadresse können nicht ermittelt werden. Stimmen die Angaben im Kanzleiprofil?", com.jdimension.jlawyer.client.utils.DesktopUtils.POPUP_TITLE_ERROR, JOptionPane.ERROR_MESSAGE);
                    return;
                }
                
                String coordinates2 = FindRouteToAddress.getCoordinates(destinationAddress);
                if(coordinates2==null) {
                    JOptionPane.showMessageDialog(this, "Koordinaten der Zieladresse können nicht ermittelt werden. Ist die Adresse des Beteiligten vollständig?", com.jdimension.jlawyer.client.utils.DesktopUtils.POPUP_TITLE_ERROR, JOptionPane.ERROR_MESSAGE);
                    return;
                }
```
also, for displaying error messages, use the following method which will take care of displaying the right icon depending on the severity of the error:

`JOptionPane.showMessageDialog(this, "Koordinaten der Zieladresse können nicht ermittelt werden. Ist die Adresse des Beteiligten vollständig?", com.jdimension.jlawyer.client.utils.DesktopUtils.POPUP_TITLE_ERROR, JOptionPane.ERROR_MESSAGE);`